### PR TITLE
Unknown attributes

### PR DESCRIPTION
**Pull request by mstimberg**
Ready to merge from my side, see discussion in #393.

### CODE REVIEW

File: brian2/tests/test_neurongroup.py [hunk 1]
```diff
@@ -856,6 +856,23 @@ def test_state_variable_access_strings():
...
 
+@attr('codegen-independent')
+def test_unknown_state_variables():
+    # Test how setting attribute names that do not correspond to a state
+    # variable are handled
+    G = NeuronGroup(10, 'v : 1')
+    assert_raises(AttributeError, lambda: setattr(G, 'unknown', 42))
+
+    # Creating a new private attribute should be fine
+    G._unknown = 42
```
Comment (thesamovar) [inline brian2/tests/test_neurongroup.py]: I approve of this magic constant.

**Comment by thesamovar** [comment]
Code looks good and tests pass on my machine, I'm happy to merge - but the Travis build failed.

**Comment by thesamovar** [comment]
Seems like a trivial doctest problem in Python 3.x (string representation of dicts has changed). Simply removing those doctests should fix it I think. Feel free to do what you think is best and merge after.

**Comment by mstimberg** [comment]
I converted the doctest into a normal test, travis tests are passing now.

**Pull request merged**